### PR TITLE
Change ReconcileAction to Action and add associated ctors

### LINKED ISSUE #317: ReconcilerAction should be an enum
IMHO `ReconcilerAction` should be an enum, rather than a struct. This would handling the outcome much simpler.

I would imagine something like:

~~~rust
pub enum ReconcilerAction {
  Done,
  Requeue,
  RequeueAfter(Duration),
}
~~~

Or maybe:

~~~rust
pub enum ReconcilerAction {
  Done,
  Requeue(Option<Duration>),
}
~~~

This would allow you to handle outcomes like this:

~~~rust
match do_reconcile() {
  None => {},
  Requeue => {},
  RequeueAfter(duration) => {},
}
~~~

### PR DESCRIPTION
**Pull request by clux**
fixes #317

see https://github.com/kube-rs/kube-rs/issues/317#issuecomment-1070621137 for motivation.

leaves the original struct in place, but because we are renaming it, it might also make sense to make members private.

### CODE REVIEW

File: kube-runtime/src/controller/mod.rs [hunk 1]
```diff
@@ -49,12 +49,44 @@ pub enum Error<ReconcilerErr: std::error::Error + 'static, QueueErr: std::error:
...
+    /// frequent changes to the underlying object, or some other hook to retain eventual consistency.
+    pub fn await_change() -> Self {
+        Self { requeue_after: None }
+    }
+}
+
+impl Default for Action {
+    fn default() -> Self {
+        Action::requeue(Duration::from_secs(3600))
+    }
```
Comment (clux) [inline kube-runtime/src/controller/mod.rs:89]: am on the fence about including this

File: kube-runtime/src/controller/mod.rs [hunk 2]
```diff
@@ -49,12 +49,44 @@ pub enum Error<ReconcilerErr: std::error::Error + 'static, QueueErr: std::error:
...
+    ///
+    /// **Warning**: If you have watch desyncs, it is possible to miss changes entirely.
+    /// It is therefore not recommended to disable requeuing this way, unless you have
+    /// frequent changes to the underlying object, or some other hook to retain eventual consistency.
+    pub fn await_change() -> Self {
+        Self { requeue_after: None }
+    }
+}
+
+impl Default for Action {
```
Comment (teozkr) [inline kube-runtime/src/controller/mod.rs:86]: Why the `Default` instance? Why an hour in particular?

Comment (clux) [reply kube-runtime/src/controller/mod.rs:86]: It's not necessary, but it felt like a nice number to point to. Desyncs are not particularly frequent these days. Have not actually seen one in ages in my controllers, so a pretty infrequent periodic might be a sensible value.

But happy to drop this, I felt it was nice to have during docs writing, but it's hard to nail down an accurate value. Kind of depends on app characteristics.

Comment (clux) [reply kube-runtime/src/controller/mod.rs]: have removed it. can't find justifications for a default in other sources.

**Review by teozkr — APPROVED** [review]

**Pull request merged**